src/System.Runtime.Extensions/src/System/IO/TextReader.cs

```diff
@@ -95,8 +105,8 @@ public virtual int Read(char[] buffer, int index, int count)
                 {
                     break;
                 }
-                buffer[index + n++] = (char)ch;
-            } while (n < count);
+                destination[n++] = (char)ch;
+            } while (n < destination.Length);
```

Comment:
This loop looks like it'll hit an IndexOutOfRangeException if destination.Length == 0.  Is there a test for that?  In fact, on netfx this code throws an IndexOutOfRangeException:
```C#
using System.IO;

class Program
{
    static void Main()
    {
        var reader = new MyTextReader();
        reader.Read(new char[0], 0, 0);
    }
}

class MyTextReader : TextReader
{
    public override int Read() => 'a';
}
```
cc: @JeremyKuhne 

I'd prefer seeing the loop written like:
```C#
int n;
for (n = 0; n < destination.Length; n++)
{
    int ch = Read();
    if (ch == -1)
    {
        break;
    }
    destination[n] = (char)ch;
}
return n;
```
which IMHO not only makes it easier to understand, but also addresses the out-of-bounds exception.

Reply:
@stephentoub As per the above review feedback, I have to invert the logic to have ```Read(Span<char>)``` delegate to ```Read(char []...)```. Above is how it was originally in the code, you want me to make changes along with the straightening of the logic...?

Reply:
I suggest leaving the existing code for now.  I'll open an issue for the problematic behavior.  It can be addressed in a separate PR.